### PR TITLE
Offset calculation change to body margin-bottom

### PR DESCRIPTION
**Pull request by codewizz**
Hello,
this PR fixes problems related to sites that use an absolute footers. In order to better get the idea here are some tests I have made:

http://debugbar.padding.codewizz.com/ - this one is using the old padding-bottom inline style (just try to expand the debugbar, you will see the footer starts to shrink and finally hides under the debugbar panel)
http://debugbar.margin.codewizz.com/ - this one is using the new margin-bottom inline style (at first it looks like the footer is being overlapped but you can scroll down and see all the web content like it should be displayed).

In addition I have added margin-bottom style removal from body once debugbar is closed or minimized. This is especially handy when your front-ender is working on a site and you don't want to toggle debugbar on and off so he could see the webpage like it should be displayed (it was a huge problem before when debugbar was using padding-bottom inline style on a body tag).

I hope this helps!

Looking forward.

### CODE REVIEW

**Comment by codewizz** [comment]
Hello. Is there any chance these changes are going to a code base any time soon?

**Comment by xewl** [comment]
+1 to merge this. The white-space is killing my productivity by manually overwriting it every project (not significantly, but, hey...)

**Comment by codewizz** [comment]
@xewl looks like phpdebugbar is not maintained / updated for some time now. Until this PR is merged or if it's ever merged you can use the following lines in composer to have this PR applied to the codebase.

```
    "require-dev": {
        "barryvdh/laravel-debugbar": "*",
        "maximebf/debugbar": "dev-codewizz-dynamic-margin as 1.11.0"
    },
    "repositories": [{
        "type": "git",
        "url": "https://github.com/codewizz/php-debugbar"
    }],
```

you might need to set these too:
```
   "minimum-stability": "dev",
   "prefer-stable": true
```

**Comment by barryvdh** [comment]
Okay let's give this a try.

**Comment by barryvdh** [comment]
It seems to work for me, expect for the state that the only the bar is shown. So not closed, but minimized.

See 
image
When closed or fully opened, the copyright is shown, but when minimized, it isn't.

File: src/DebugBar/Resources/debugbar.js [hunk 1]
```diff
@@ -813,13 +822,17 @@ if (typeof(PhpDebugBar) == 'undefined') {
...
-         * Recomputes the padding-bottom css property of the body so
+         * Recomputes the margin-bottom css property of the body so
          * that the debug bar never hides any content
          */
         recomputeBottomOffset: function() {
-            if (this.options.bodyPaddingBottom) {
-                var height = parseInt(this.$el.height()) + this.options.bodyPaddingBottomHeight;
-                $('body').css('padding-bottom', height);
+            if (this.options.bodyMarginBottom) {
+                if (this.isMinimized() || this.isClosed()) {
```
Comment (barryvdh) [inline src/DebugBar/Resources/debugbar.js]: Should this just check isClosed() and not isMinimized()?

Comment (barryvdh) [inline src/DebugBar/Resources/debugbar.js]: https://github.com/maximebf/php-debugbar/commit/c2455dae3223ec2382f5eaa32e47b9a988e129f0

**Comment by xewl** [comment]
And suddenly, there's liveness! Nice (:

Actually.. As it's a debugbar... it's not a problem that I don't see the footer "partially".
I'd be fine with it just hovering fixed while "collapsed", but if you seemed to fix this one nonetheless, I'd be happy too.

#kudos

**Comment by barryvdh** [comment]
Yeah see the mentioned commit, https://github.com/maximebf/php-debugbar/commit/c2455dae3223ec2382f5eaa32e47b9a988e129f0

**Pull request merged**